src/Extensions/HttpContextExtensions.cs

```diff
@@ -2,17 +2,20 @@
 // The Apache v2. See License.txt in the project root for license information.
 
 using System;
+
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
+
 using Wangkanai.Detection.Models;
 
-namespace Microsoft.AspNetCore.Http
+namespace Wangkanai.Detection.Extensions
 {
     internal static class HttpContextExtensions
     {
         private const string ResponsiveContextKey = "Responsive";
 
         public static ISession? SafeSession(this HttpContext httpContext)
-            => httpContext.Features.Get<ISessionFeature>() == null
+            => httpContext.Features.Get<ISessionFeature?>() == null
```

Comment:
How could a generic type be null? very interesting concept. I have never seen such usage (don't know it existed too). 

Thank you very much for the addition to the code and my knowledge.

Reply:
Nullable reference types is just a static analyzer hints, so technically nothing changed, it just allows compiler to think that Features.Get<T>() resolution could lead to null. In this case it's an interface which is always reference type independent of implementation (struct or class), in case of struct it would be boxed into object. BTW all reference types could be nullable by definition and so this question mark compiler hints always allowed.